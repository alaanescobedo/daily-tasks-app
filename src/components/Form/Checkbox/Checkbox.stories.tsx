import { ComponentStory, ComponentMeta } from '@storybook/react'
import { Checkbox, InputCheckboxProps } from './Checkbox'

const InputCheckboxStory: ComponentMeta<typeof Checkbox> = {
  title: 'Inputs/Checkbox',
  component: Checkbox
}
export default InputCheckboxStory

const Template: ComponentStory<typeof Checkbox> = (args) => <Checkbox {...args} />

export const InputCheckbox = Template.bind({})
const checkboxArgs: InputCheckboxProps = {
  checked: true,
  id: 'recurrent',
  handleCheckbox: () => { }
}
InputCheckbox.args = checkboxArgs
