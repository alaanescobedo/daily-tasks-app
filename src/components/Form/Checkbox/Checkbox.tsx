import { Icon } from '@components/Icons'
import sharedStyles from '../shared/styles.module.css'

export interface InputCheckboxProps {
  id: string
  checked: boolean
  handleCheckbox?: (id: string) => void
  idIcon?: string
}

export const Checkbox = ({ id, checked, handleCheckbox, idIcon = 'recurrent-icon' }: InputCheckboxProps): JSX.Element => {
  return (
    <span className={sharedStyles.input_recurrent} onClick={() => (handleCheckbox != null) ? handleCheckbox(id) : null}>
      {checked
        ? <Icon id={idIcon} icon='check' />
        : null}
      <input name={id} id={id} type='checkbox' className={sharedStyles.checkbox} checked={checked} readOnly />
    </span>
  )
}
