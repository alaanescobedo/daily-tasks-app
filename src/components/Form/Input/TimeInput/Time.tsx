import sharedStyles from '../../shared/styles.module.css'

interface Props {
  id: string
  defaultValue: string
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export const Time = ({ id, defaultValue, handleChange }: Props): JSX.Element => {
  return (
    <input name={id} id={id} type='time' className={sharedStyles.input_date_hour} onChange={handleChange} defaultValue={defaultValue} />
  )
}
