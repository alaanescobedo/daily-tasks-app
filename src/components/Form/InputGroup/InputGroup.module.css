.form_group {
  display: flex;
  flex-direction: column;
  gap: .4rem;
}

.form_group label {
  color: rgb(var(--highlight));
  font-size: 1.4rem;
  font-weight: 600;
}