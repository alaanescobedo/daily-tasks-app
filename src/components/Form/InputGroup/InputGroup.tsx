import styles from './InputGroup.module.css'

export interface InputGroupProps {
  children: React.ReactNode
  label?: string
  htmlFor: string
  className?: string
}

export const InputGroup = ({ children, label, htmlFor, className }: InputGroupProps): JSX.Element => {
  return (
    <div className={className ?? styles.form_group}>
      {children}
      <label htmlFor={htmlFor}>{label}</label>
    </div>
  )
}
