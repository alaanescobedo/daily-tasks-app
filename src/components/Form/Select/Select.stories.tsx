import { ComponentStory, ComponentMeta } from '@storybook/react'

import { Select, InputSelectProps } from './Select'

const InputSelectStory: ComponentMeta<typeof Select> = {
  title: 'Inputs/Select',
  component: Select
}
export default InputSelectStory

const Template: ComponentStory<typeof Select> = (args) => <Select {...args} />

export const PrioritySelect = Template.bind({})
const prioritySelectArgs: InputSelectProps = {
  id: 'priority',
  defaultValue: 'medium',
  handleChange: () => { },
  options: [
    { id: 'low', label: 'Low' },
    { id: 'medium', label: 'Medium' },
    { id: 'high', label: 'High' }
  ]
}
PrioritySelect.args = prioritySelectArgs

export const DateSelect = Template.bind({})
const dateSelectArgs: InputSelectProps = {
  id: 'date',
  defaultValue: 'friday',
  handleChange: () => { },
  options: [
    { id: 'friday', label: 'Friday' },
    { id: 'saturday', label: 'Saturday' },
    { id: 'sunday', label: 'Sunday' },
    { id: 'monday', label: 'Monday' },
    { id: 'tuesday', label: 'Tuesday' },
    { id: 'wednesday', label: 'Wednesday' },
    { id: 'thursday', label: 'Thursday' }
  ]
}
DateSelect.args = dateSelectArgs
