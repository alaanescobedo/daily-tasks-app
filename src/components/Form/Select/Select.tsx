import { ChangeEvent } from 'react'
import sharedStyles from '@components/Form/shared/styles.module.css'

export type PriorityId = 'low' | 'medium' | 'high'
export type WeekdaysId = 'monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday'

export type WeekdaysLabel = 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday' | 'Friday' | 'Saturday' | 'Sunday' | 'Today' | 'Tomorrow'
export type PriorityLabel = 'Low' | 'Medium' | 'High'

export interface InputSelectProps {
  id: string
  options: Array<{ id: PriorityId | WeekdaysId, label: PriorityLabel | WeekdaysLabel }>
  handleChange: (e: ChangeEvent<HTMLSelectElement>) => void
  defaultValue: string
}

export const Select = ({ id, options, handleChange, defaultValue }: InputSelectProps): JSX.Element => {
  return (
    <select name={id} id={id} className={`${sharedStyles.input_select}`} onChange={handleChange} defaultValue={defaultValue}>
      {options.map((option, index) => (
        <option key={index} value={option.id}>{option.label}</option>
      ))}
    </select>
  )
}
