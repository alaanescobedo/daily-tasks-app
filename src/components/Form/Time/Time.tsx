import { getCurrentDate } from '@utils/index'
import styles from '@components/Form/shared/styles.module.css'

export interface InputTimeProps {
  id: string
  defaultValue: string
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void
}

export const Time = ({ id, defaultValue, handleChange }: InputTimeProps): JSX.Element => {
  const minTime = getCurrentDate().split(',')[2].split(':').slice(0, 2).join(':').trim()
  return (
    <input name={id} id={id} type='time' className={styles.input_date_hour} onChange={handleChange} defaultValue={defaultValue} min={minTime} />
  )
}
