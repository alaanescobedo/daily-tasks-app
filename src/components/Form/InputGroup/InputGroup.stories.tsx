import { ComponentStory, ComponentMeta } from '@storybook/react'
import { InputSelectProps } from '../Select/Select'
import { DateSelect, PrioritySelect } from '../Select/Select.stories'

import { InputGroup, InputGroupProps } from './InputGroup'
import { InputTime } from '../Time/Time.stories'
import { InputTimeProps } from '../Time/Time'
import { InputCheckbox } from '../Checkbox/Checkbox.stories'
import { InputCheckboxProps } from '../Checkbox/Checkbox'
import styles from '@views/NewTask/NewTask.module.css'

const InputGroupStory: ComponentMeta<typeof InputGroup> = {
  title: 'Inputs/InputGroup',
  component: InputGroup
}
export default InputGroupStory

const Template: ComponentStory<typeof InputGroup> = (args) => <InputGroup {...args} />

export const GroupDate = Template.bind({})
const groupDateArgs: InputGroupProps = {
  children: <DateSelect {...DateSelect.args as InputSelectProps} />,
  label: 'Date',
  htmlFor: 'date',
  className: styles.group_date_day
}
GroupDate.args = groupDateArgs

export const GroupHour = Template.bind({})
const groupHourArgs: InputGroupProps = {
  children: <InputTime {...InputTime.args as InputTimeProps} />,
  label: 'Hour',
  htmlFor: 'time',
  className: styles.group_date_hour
}
GroupHour.args = groupHourArgs

export const GroupCheckbox = Template.bind({})
const groupCheckboxArgs: InputGroupProps = {
  children: <InputCheckbox {...InputCheckbox.args as InputCheckboxProps} />,
  label: 'Recurrent',
  htmlFor: 'recurrent',
  className: styles.group_recurrent
}
GroupCheckbox.args = groupCheckboxArgs

export const GroupPriority = Template.bind({})
const groupPriorityArgs: InputGroupProps = {
  children: <PrioritySelect {...PrioritySelect.args as InputSelectProps} />,
  label: 'Priority',
  htmlFor: 'priority',
  className: styles.group_priority
}
GroupPriority.args = groupPriorityArgs
