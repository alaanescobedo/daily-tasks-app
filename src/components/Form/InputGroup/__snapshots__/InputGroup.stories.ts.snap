// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Form/InputGroup Group Checkbox 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <div
      className="group_recurrent"
    >
      <span
        className="input_recurrent"
        onClick={[Function]}
      >
        <svg
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
          />
        </svg>
        <input
          checked={true}
          className="checkbox"
          id="recurrent"
          name="recurrent"
          readOnly={true}
          type="checkbox"
        />
      </span>
      <label
        htmlFor="recurrent"
      >
        Recurrent
      </label>
    </div>
     
  </div>,
  " ",
]
`;

exports[`Storyshots Form/InputGroup Group Date 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <div
      className="group_date_day"
    >
      <select
        className="input_select"
        defaultValue="friday"
        id="date"
        name="date"
        onChange={[Function]}
      >
        <option
          value="friday"
        >
          Friday
        </option>
        <option
          value="saturday"
        >
          Saturday
        </option>
        <option
          value="sunday"
        >
          Sunday
        </option>
        <option
          value="monday"
        >
          Monday
        </option>
        <option
          value="tuesday"
        >
          Tuesday
        </option>
        <option
          value="wednesday"
        >
          Wednesday
        </option>
        <option
          value="thursday"
        >
          Thursday
        </option>
      </select>
      <label
        htmlFor="date"
      >
        Date
      </label>
    </div>
     
  </div>,
  " ",
]
`;

exports[`Storyshots Form/InputGroup Group Hour 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <div
      className="group_date_hour"
    >
      <input
        className="input_date_hour"
        defaultValue="12:00"
        id="time"
        min="12:00"
        name="time"
        onChange={[Function]}
        type="time"
      />
      <label
        htmlFor="time"
      >
        Hour
      </label>
    </div>
     
  </div>,
  " ",
]
`;

exports[`Storyshots Form/InputGroup Group Priority 1`] = `
Array [
  " ",
  <div
    style={
      Object {
        "--background": "2, 40, 60",
        "--highlight": "255,  202,  16",
        "--opacity-1": 0.1,
        "--opacity-7": 0.7,
        "--primary": "206, 180, 249",
        "--secondary": "189, 231, 255",
        "--tertiary": "176, 239, 165",
        "--text": "245, 245, 245",
        "0.1": "--opacity-1",
        "0.7": "--opacity-7",
      }
    }
  >
     
    <div
      className="group_priority"
    >
      <select
        className="input_select"
        defaultValue="medium"
        id="priority"
        name="priority"
        onChange={[Function]}
      >
        <option
          value="low"
        >
          Low
        </option>
        <option
          value="medium"
        >
          Medium
        </option>
        <option
          value="high"
        >
          High
        </option>
      </select>
      <label
        htmlFor="priority"
      >
        Priority
      </label>
    </div>
     
  </div>,
  " ",
]
`;
